Portuguese translation what it is to be a maintainer (#1118)
ropensci/roweb3: content/blog/2023-02-07-what-does-it-mean-to-maintain-a-package/index.pt.md

---
title: O que significa manter um pacote?
author:
- Maëlle Salmon
date: '2023-02-07'

> slug: what-does-it-mean-to-maintain-a-package

Author: ```suggestion
slug: o-que-significa-manter-um-pacote
```

categories:
- blog
tags:
- package
- maintenance
package_version: 0.1.0
description: Uma tentativa de definir o que significa a manutenção de pacotes, com
  algumas dicas.
params:
  doi: 10.59350/vv6xb-53139
---

> Parte do que pretendemos fazer na rOpenSci é cultivar uma comunidade de mantenedores de pacotes que se ajudam mutuamente.

Reviewer: ```suggestion
Parte do que pretendemos fazer na rOpenSci é cultivar uma comunidade de pessoas mantenedoras de pacotes que se ajudam mutuamente.
```

> Além do suporte durante a manutenção do pacote, também queremos dar suporte aos mantenedores que desejam seguir em frente. As situações podem mudar e pode chegar um momento em que um mantenedor queira passar a manutenção para outro. Se um mantenedor se encontrar nessa situação e quiser transferir a manutenção, nós o ajudaremos da seguinte forma [publicidade](/blog/2022/10/17/maintain-or-co-maintain-an-ropensci-package/) e também ajudamos o(s) novo(s) mantenedor(es) com conselhos, geralmente do tipo "olhe ao redor para ver se há algo que precise de conserto e, em seguida, faça a manutenção de rotina".

Reviewer: ```suggestion
Além do suporte durante a manutenção do pacote, também queremos dar suporte às pessoas mantenedores que desejam seguir em frente. As situações podem mudar e pode chegar um momento em que uma pessoa que mantém o pacote queira passar a manutenção para outra. Se uma pessoa mantenedora se encontrar nessa situação e quiser transferir a manutenção, nós ajudaremos [divulgando](/blog/2022/10/17/maintain-or-co-maintain-an-ropensci-package/) e também ajudamos a(s) nova(s) pessoa(s) mantenedora(s) com conselhos, geralmente do tipo "olhe ao redor para ver se há algo que precise de conserto e, em seguida, faça a manutenção de rotina".
```

Mas o que é manutenção de rotina? Esta postagem é uma tentativa de definir o que significa manutenção de pacotes, com algumas dicas.

> ## Manutenção de pacotes como propriedade

Reviewer: ```suggestion
## Manutenção de pacotes como compromisso de quem assume o pacote
```

> Como mantenedor de pacotes (ou equipe de mantenedores),

Reviewer: ```suggestion
Como mantenedor(a) de pacotes (ou equipe de pessoas mantenedoras),
```

- você é responsável pelo escopo do pacote: Você quer adicionar um recurso? A decisão é sua!

> - você está encarregado de planejar o trabalho na base de código: Você deve pensar em melhorias e organizá-las em marcos.

Reviewer: ```suggestion
- você define o planejamento do trabalho na base de código: você deve pensar em melhorias e organizá-las em marcos.
```

> - você está no controle da qualidade do código: Você quer dedicar mais tempo a esse pull request? Ou você aceitará e acumulará um pouco de [dívida técnica](https://en.wikipedia.org/wiki/Technical_debt)? Novamente, a decisão é sua!

Reviewer: ```suggestion
- você está no controle da qualidade do código: Você quer dedicar mais tempo a esse *pull request*? Ou você aceitará e acumulará um pouco de [dívida técnica](https://en.wikipedia.org/wiki/Technical_debt)? Novamente, a decisão é sua!
```

No entanto, você não possui apenas esses aspectos técnicos e de produtividade.
Em nossa chamada da comunidade anterior sobre [Manutenção de um pacote R](/commcalls/2020-03-18/) Erin Grand definiu a manutenção de um pacote como "propriedade em torno da comunidade do pacote".

> Que ótima maneira de reconhecer a *pessoas* que criam e usam um software!

Reviewer: ```suggestion
Que ótima maneira de reconhecer as *pessoas* que criam e usam um software!
```

- Você não precisa resolver todos os problemas assim que eles surgirem (além dos prazos rigorosos do CRAN, é claro). Você pode planejar períodos de atividade e inatividade no seu repositório, possivelmente indicando isso claramente em um problema fixado ou na documentação. O manual de metas explica [períodos de ausência do escritório](https://books.ropensci.org/targets/help.html#out-of-office) mas a inatividade também pode muito bem ser o estado mais comum do seu pacote, com atividade em rajadas. Seu pacote, sua escolha!
- Você pode ajustar suas diretrizes de contribuição ao longo do tempo. Não é vergonha nenhuma pedir a um usuário que forneça uma resposta em vez de passar horas tentando adivinhar o significado do texto do problema. Por meio de diretrizes de contribuição e explicações gentis, você pode transferir perguntas gerais sobre o seu pacote para um local onde seja mais fácil para você, por exemplo, GitHub Discussions em vez de Issues.

> - Certifique-se de receber as notificações (seu endereço de e-mail correto está listado, você observa seu próprio repositório?) Você não precisa ler ou agir sobre elas imediatamente, mas é bom não perdê-las.

Reviewer: ```suggestion
- Verifique se você realmente está recebendo notificações (o e-mail listado está correto? a opção de acompanhar o repositório (*watch*) está ativada?). Não é necessário ler ou agir imediatamente, mas é importante não perder os avisos.
```

- Você pode escolher onde seu pacote será publicado. Se as regras de um repositório de publicação forem uma fonte de muita dor em comparação com a facilidade de distribuição, você pode optar por deixá-lo.

> - Você também pode tentar obter financiamento para o seu trabalho. Consulte, por exemplo, a seção [R Consortium call for proposals](https://www.r-consortium.org/all-projects/call-for-proposals) (duas vezes por ano).

Reviewer: ```suggestion
- Você também pode tentar obter financiamento para o seu trabalho. Consulte, por exemplo, a [chamada de propostas da R Consortium](https://www.r-consortium.org/all-projects/call-for-proposals) (duas vezes por ano).
```

> ## Manutenção de pacotes como serviço de limpeza

Reviewer: ```suggestion
## Manutenção de pacotes como "arrumação da casa"
```
